sources/packages/backend/libs/utilities/src/parallel-processing-utils.ts

```diff
@@ -0,0 +1,41 @@
+import * as os from "os";
+
+/**
+ * Execute processes in parallel during processing of high volume
+ * of records to improve the performance in a controlled fashion.
+ * @param createPromise Implementation that calls processing function which
+ * returns a promise.
+ * @param inputs input data from which the bulk processing is done.
+ * @param maxParallelRequests  Maximum number of parallel processes allowed to be executed at same time.
+ * @returns resolved responses of promise.
+ */
+export const processRequestsInParallel = async <P, I>(
+  createPromise: (record: I) => Promise<P>,
```

Comment:
maybe it would be good to rename `record` to `input` as `createPromise: (input: I) => Promise<P>`